sdk/Sdk/Targets/Microsoft.Azure.Functions.Worker.Sdk.Publish.targets

```diff
@@ -21,4 +21,28 @@ WARNING:  DO NOT MODIFY this file unless you are knowledgeable about MSBuild and
   <Import Project="$(MSBuildThisFileDirectory)Microsoft.Azure.Functions.Worker.Sdk.Publish.$(PublishProtocol).targets"  
           Condition="Exists('$(MSBuildThisFileDirectory)Microsoft.Azure.Functions.Worker.Sdk.Publish.$(PublishProtocol).targets')" />
 
+    <Target
+          Name="AssignFunctionsBaseImage"
+          BeforeTargets="ComputeContainerBaseImage"
+          DependsOnTargets="_FunctionsPreBuild" >
+        <PropertyGroup>
+            <_FunctionsRuntimeMajorVersion>$(_AzureFunctionsVersionStandardized.Split('-')[0].TrimStart('vV'))</_FunctionsRuntimeMajorVersion>
+            <ContainerBaseImage Condition="'$(ContainerBaseImage)' == ''">mcr.microsoft.com/azure-functions/dotnet-isolated:$(_FunctionsRuntimeMajorVersion)-dotnet-isolated$(TargetFrameworkVersion.TrimStart('vV'))</ContainerBaseImage>
+            <ContainerWorkingDirectory Condition="'$(ContainerWorkingDirectory)' == ''">/home/site/wwwroot</ContainerWorkingDirectory>
+            <!-- Functions base images only support amd64 runtimes -->
+            <ContainerRuntimeIdentifier Condition="'$(ContainerRuntimeIdentifier)' == ''">linux-x64</ContainerRuntimeIdentifier>
+        </PropertyGroup>
+        <ItemGroup Condition="'$(FunctionsContainerOmitDefaultEnvVars)' != 'true'">
```

Comment:
In the issue description you mention wanting to defensively set this value, i.e. only setting the value if `AzureWebJobScriptRoot` is unset by a user at this point. You can check if the Item is already present by checking for an Item with that Identity:

```xml
<ContainerEnvironmentVariable
                    Condition="@(ContainerEnvironmentVariable->AnyHaveMetadataValue('Identity', 'AzureWebJobsScriptRoot')) == false"
                    Include="AzureWebJobsScriptRoot"
                    Value="$(ContainerWorkingDirectory)" />
```

Reply:
I like this MUCH better. Added to newer iteration. Thanks!